File: java/common/src/main/java/com/ais/middleware/common/events/integration/PolicyAdminSystemResponseReceivedEvent.java
package com.ais.middleware.common.events.integration;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.OffsetDateTime;
import java.util.List;

/**
 * Published once by Platform.Integration after a successful PAS call.
 * It is a pub/sub fan-out event consumed simultaneously by PolicyIssuanceAndLifecycleManagement, BillingAndFinanceManagement, and CustomerIdentityAndRelationshipManagement, and no domain re-publishes or forwards it under the Single Publisher Rule.
 */
public record PolicyAdminSystemResponseReceivedEvent(
        String issuanceId,
        String targetPas,
        String accountServiceRequestNumber,
        List<String> policyNumbers,
        OffsetDateTime receivedAt
) {
    @JsonCreator
    public PolicyAdminSystemResponseReceivedEvent(
            @JsonProperty("issuanceId") String issuanceId,
            @JsonProperty("targetPas") String targetPas,
            @JsonProperty("accountServiceRequestNumber") String accountServiceRequestNumber,
            @JsonProperty("policyNumbers") List<String> policyNumbers,
            @JsonProperty("receivedAt") OffsetDateTime receivedAt
    ) {
        this.issuanceId = issuanceId;
        this.targetPas = targetPas;
        this.accountServiceRequestNumber = accountServiceRequestNumber;
        this.policyNumbers = policyNumbers;
        this.receivedAt = receivedAt;
    }
}

File: dotnet/Middleware.Contracts/Events/PolicyAdminSystemResponseReceivedEvent.cs
using NServiceBus;

namespace Middleware.Contracts.Events;

public sealed class PolicyAdminSystemResponseReceivedEvent : IEvent
{
    public string IssuanceId { get; set; } = string.Empty;
    public string AccountId { get; set; } = string.Empty;
    public string TargetPas { get; set; } = string.Empty;
    public string AccountServiceRequestNumber { get; set; } = string.Empty;
    public List<string> PolicyNumbers { get; set; } = [];
    public DateTimeOffset ReceivedAt { get; set; } = DateTimeOffset.UtcNow;
}
